src/Nox.Types/Types/VatNumber/VatNumber.cs

```diff
@@ -1,9 +1,125 @@
+(BE)?0[0-9]{9} |                              # Belgium
+(BG)?[0-9]{9,10} |                            # Bulgaria
+(CY)?[0-9]{8}L |                              # Cyprus
+(CZ)?[0-9]{8,10} |                            # Czech Republic
+(DE)?[0-9]{9} |                               # Germany
+(DK)?[0-9]{8} |                               # Denmark
+(EE)?[0-9]{9} |                               # Estonia
+(EL|GR)?[0-9]{9} |                            # Greece
+(ES)?[0-9A-Z][0-9]{7}[0-9A-Z] |               # Spain
+(FI)?[0-9]{8} |                               # Finland
+(FR)?[0-9A-Z]{2}[0-9]{9} |                    # France
+(GB)?([0-9]{9}([0-9]{3})?|[A-Z]{2}[0-9]{3}) | # United Kingdom
+(HU)?[0-9]{8} |                               # Hungary
+(IE)?[0-9]S[0-9]{5}L |                        # Ireland
+(IT)?[0-9]{11} |                              # Italy
+(LT)?([0-9]{9}|[0-9]{12}) |                   # Lithuania
+(LU)?[0-9]{8} |                               # Luxembourg
+(LV)?[0-9]{11} |                              # Latvia
+(MT)?[0-9]{8} |                               # Malta
+(NL)?[0-9]{9}B[0-9]{2} |                      # Netherlands
+(PL)?[0-9]{10} |                              # Poland
+(PT)?[0-9]{9} |                               # Portugal
+(RO)?[0-9]{2,10} |                            # Romania
+(SE)?[0-9]{12} |                              # Sweden
+(SI)?[0-9]{8} |                               # Slovenia
+(SK)?[0-9]{10}                                # Slovakia
+)$", 
+        RegexOptions.IgnorePatternWhitespace | RegexOptions.IgnoreCase,
+        TimeSpan.FromSeconds(1));
+
+    #endregion
+
+    private VatNumberOptions _typeOptions = new();
+
+    public string VatNumberValue
+    {
+        get => Value.VatNumberValue;
+        private set => Value = (value, Value.CountryCode);
+    }
+
+    public CountryCode2 CountryCode
+    {
+        get => Value.CountryCode;
+        private set => Value = (Value.VatNumberValue, value);
+    }
+
     /// <summary>
-    /// Represents a Nox <see cref="VatNumber"/> type and value object.
+    /// Initializes a new instance of the <see cref="VatNumber"/> class with default values.
     /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class VatNumber : ValueObject<(string VatNumber, CountryCode2 countryCode2), VatNumber>
+    public VatNumber() { }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="VatNumber"/> class with the specified values.
+    /// </summary>
+    /// <param name="value">The VAT number value.</param>
+    /// <param name="countryCode">The two-letter country codes (ISO alpha-2).</param>
+    /// <returns>A new instance of the <see cref="VatNumber"/> class.</returns>
+    public static VatNumber From(string value, CountryCode2 countryCode) => 
+        From((value, countryCode));
+
+    public static VatNumber From(string value, VatNumberOptions typeOptions) =>
```

Comment:
for the From Overloads, I would suggest

```
From(string vatNumber, string countryCode2) => From(vatNumber , new VatNumberOptions(){CountryCode2 = countryCode2})
From(string vatNumber, VatNumberOptions typeOptions)

```
Do not combine VatNumberOptions  and CountryCode, because they may contradict, and that is not the intention

Reply:
I've updated the overloads to the suggested ones. But I have a small question about it - don't we expect that the user will use DefaultCountryCode and then insert VatNumber with other CountryCode?

Reply:
If I understand the question, the user needs to add a VatNumber and express in what countryCode, via Options or explicit parameter, the default is just a facility to not express the country code and use the default in the options. We could also create a overload From (VatNumber), that would use the default from options.